src/helperFunctions/yara_binary_search.py

```diff
@@ -75,25 +85,37 @@ def get_binary_search_result(self, task: Tuple[bytes, Optional[str]]):
             except CalledProcessError as process_error:
                 return 'Error when calling YARA:\n{}'.format(process_error.output.decode())
 
-    def _get_raw_result(self, firmware_uid, temp_rule_file):
+    def _get_raw_result(self, firmware_uid: Optional[str], temp_rule_file: NamedTemporaryFile) -> str:
         if firmware_uid is None:
             raw_result = self._execute_yara_search(temp_rule_file.name)
         else:
             raw_result = self._execute_yara_search_for_single_firmware(temp_rule_file.name, firmware_uid)
         return raw_result
 
     @staticmethod
-    def _prepare_temp_rule_file(temp_rule_file, yara_rules):
+    def _prepare_temp_rule_file(temp_rule_file: NamedTemporaryFile, yara_rules: bytes):
         compiled_rules = yara.compile(source=yara_rules.decode())
         compiled_rules.save(file=temp_rule_file)
         temp_rule_file.flush()
 
 
-def is_valid_yara_rule_file(rules_file):
-    return get_yara_error(rules_file) is None
+def is_valid_yara_rule_file(yara_rules: Union[str, bytes]) -> bool:
+    '''
+    Check if ``yara_rules`` is a valid set of yara rules.
+
+    :param: A string containing yara rules.
+    :return: ``True`` if the rules are valid and ``False`` otherwise.
+    '''
+    return get_yara_error(yara_rules) is None
+
 
+def get_yara_error(rules_file: Union[str, bytes]) -> Optional[Exception]:
+    '''
+    Get the exception that is caused by trying to compile ``rules_file`` with yara or ``None`` if there is none.
 
-def get_yara_error(rules_file):
+    :param rules_file: A string containing yara rules.
+    :result: The exception if compiling the rules causes an exception or ``None`` otherwise.
+    '''
     if isinstance(rules_file, bytes):
         rules_file = rules_file.decode()
     try:
```

Comment:
Is there a specific reason why you haven't documented `class YaraBinarySearchScannerDbInterface` ?

Reply:
Yes: it is only used internally by the other functions (I could also have renamed it to `_YaraBinarySearchScannerDbInterface` but that looked ugly)

Reply:
Haha. Yes it does